### PR TITLE
feat: [CI-5075]: Added manager changes for detach step - K8 & VM

### PR DESCRIPTION
**Pull request by raghavharness**
## Describe your changes
Added support for background step in CI
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36265)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 9 Issues 
* Code issue 8 Code issues 
* Anti-pattern 1 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36265)

**Comment by raghavharness** [comment]
trigger ft1

**Comment by raghavharness** [comment]
trigger ft2

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java [hunk 1]
```diff
@@ -214,9 +216,16 @@ private List<ContainerDefinitionInfo> handleParallelStep(ExecutionWrapperConfig
       return containerDefinitionInfos;
     }
 
-    int steps = parallelStepElementConfig.getSections().size();
-    Integer extraMemoryPerStep = extraMemory / steps;
-    Integer extraCPUPerStep = extraCPU / steps;
+    Integer extraMemoryPerStep = 0;
+    Integer extraCPUPerStep = 0;
+
+    int steps =
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java:222]: steps --> stepsCount

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 25 of 32 files at r1, 6 of 6 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36265#-:-NAJ_gqn7uidyd609Rxk:bnfp4nl)* status: Waiting on team(s): sto.

**Review by harshjain12 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36265#-:-NAJ_vkI60jU6ouTOtCR:bnfp4nl)* status: Waiting on team(s): sto.

**Comment by raghavharness** [comment]
trigger ti1, ti2, ti4

**Comment by raghavharness** [comment]
trigger ti2, ti4

**Comment by raghavharness** [comment]
trigger ti0

**Comment by raghavharness** [comment]
trigger ti0

**Pull request merged**